query-engine/connector-test-kit-rs/query-test-macros/src/ensure_db_names.rs

```diff
@@ -0,0 +1,57 @@
+use std::collections::HashSet;
+use std::sync::Mutex;
+
+// Ensures unique test database names.
+// Reusing the same name more than once causes random test failures, retries and overall flakiness.
+pub struct UniqueTestDatabaseNames {
+    names: Mutex<HashSet<String>>,
+}
+
+impl UniqueTestDatabaseNames {
+    pub fn new() -> Self {
+        Self {
+            names: Mutex::new(HashSet::new()),
+        }
+    }
+
+    pub fn ensure_unique(
+        self: &UniqueTestDatabaseNames,
+        test_database_name: &String,
+        suite_name: &String,
+        test_name: &String,
```

Comment:
```suggestion
        test_database_name: &str,
        suite_name: &str,
        test_name: &str,
```

There's no need to require a reference to `String` specifically in this case, we can make use of the deref coercion to make it a little more idiomatic and flexible.